brillig_bytecode/src/lib.rs

```diff
@@ -477,258 +482,373 @@ mod tests {
-            memory_map: BTreeMap::from([(0 as usize, Value::from(5u128)), (1, Value::from(6u128))]),
-        };
-        initial_memory.insert(Value::from(5u128), initial_heap);
-
+            Value::from(1u128),
+            Value::from(2u128),
+            Value::from(3u128),
+            Value::from(4u128),
+        ];
+        assert_eq!(memory, expected);
+    
+        let memory = brillig_recursive_write_memory(vec![Value::from(0u128); 1024]);
+        let expected: Vec<Value> = (0..1024).map(|i| Value::from(i as u128)).collect();
+        assert_eq!(memory, expected);
+    }
+    
+    fn empty_registers() -> Registers {
+        Registers::load(vec![Value::from(0u128); 16])
+    }
+    /// Helper to execute brillig code
+    fn brillig_execute(memory: Vec<Value>, opcodes: Vec<Opcode>) -> VM {
         let mut vm = VM::new(
-            input_registers,
-            initial_memory,
-            vec![equal_cmp_opcode, store_opcode, jump_opcode, jump_if_opcode],
+            empty_registers(),
+            memory,
+            opcodes,
+            None,
         );
-
-        // equal_cmp_opcode
-        let status = vm.process_opcode();
-        assert_eq!(status, VMStatus::InProgress);
-
-        let output_cmp_value = vm.registers.get(RegisterMemIndex::Register(RegisterIndex(2)));
-        assert_eq!(output_cmp_value, Value::from(true));
-
-        // store_opcode
-        let status = vm.process_opcode();
-        assert_eq!(status, VMStatus::InProgress);
-
-        let mem_array = vm.memory[&Value::from(5u128)].clone();
-        assert_eq!(mem_array.memory_map[&3], Value::from(true));
-
-        // jump_opcode
-        let status = vm.process_opcode();
-        assert_eq!(status, VMStatus::InProgress);
-
-        // jump_if_opcode
-        let status = vm.process_opcode();
-        assert_eq!(status, VMStatus::Halted);
-
-        vm.finish();
+        loop {
+            let status = vm.process_opcode();
+            if status == VMStatus::Finished || status == VMStatus::OracleWait {
+                break;
+            }
+            assert_eq!(status, VMStatus::InProgress)
+        }
+        assert_eq!(vm.call_stack, vec![]);
+        vm
     }
 
     #[test]
-    fn oracle_array_output() {
+    fn oracle_output() {
         use crate::opcodes::OracleInput;
 
-        let input_registers = Registers::load(vec![
-            Value::from(2u128),
-            Value::from(2u128),
-            Value::from(0u128),
-            Value::from(5u128),
-            Value::from(0u128),
-            Value::from(6u128),
-            Value::from(0u128),
-        ]);
-
-        let oracle_input =
-            OracleInput::RegisterMemIndex(RegisterMemIndex::Register(RegisterIndex(0)));
-        let oracle_output =
-            OracleOutput::Array { start: RegisterMemIndex::Register(RegisterIndex(3)), length: 2 };
+        let oracle_inputs = vec![OracleInput::RegisterIndex(RegisterIndex(0))];
+        let oracle_outputs = vec![
```

Comment:
I noticed you got rid of `OracleInput::Array` and OracleOutput::Array` in the `process_opcodes` method. I was suspicious this would force you to make every oracle input and output be a register rather than a part of memory. Is this intentional? I feel like oracle array input and output should still be a part of memory. Is there some way to incorporate usage of load and store before an oracle opcode is called?

Reply:
The intention was you could just pass a memory pointer as input or output